File: servidor/src/main/java/com/taller/autos/AutoElectrico.java
package com.taller.autos;

import java.time.LocalDateTime;

public class AutoElectrico {
    private int id;
    private String marca;
    private String modelo;
    private int anio;
    private double autonomiaKm;
    private LocalDateTime fechaRegistro;
    private Bateria bateria;

    public AutoElectrico() {}

    public AutoElectrico(int id, String marca, String modelo, int anio, double autonomiaKm,
                         LocalDateTime fechaRegistro, Bateria bateria) {
        this.id = id;
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.autonomiaKm = autonomiaKm;
        this.fechaRegistro = fechaRegistro;
        this.bateria = bateria;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getMarca() { return marca; }
    public void setMarca(String marca) { this.marca = marca; }

    public String getModelo() { return modelo; }
    public void setModelo(String modelo) { this.modelo = modelo; }

    public int getAnio() { return anio; }
    public void setAnio(int anio) { this.anio = anio; }

    public double getAutonomiaKm() { return autonomiaKm; }
    public void setAutonomiaKm(double autonomiaKm) { this.autonomiaKm = autonomiaKm; }

    public LocalDateTime getFechaRegistro() { return fechaRegistro; }
    public void setFechaRegistro(LocalDateTime fechaRegistro) { this.fechaRegistro = fechaRegistro; }

    public Bateria getBateria() { return bateria; }
    public void setBateria(Bateria bateria) { this.bateria = bateria; }
}

File: cliente/AutosCliente/Models/AutoElectrico.cs
using System;

namespace AutosCliente.Models
{
    public class Bateria
    {
        public int Id { get; set; }
        public string Tipo { get; set; } = "";
        public double CapacidadKwh { get; set; }
        public int CiclosVida { get; set; }
    }

    public class AutoElectrico
    {
        public int Id { get; set; }
        public string Marca { get; set; } = "";
        public string Modelo { get; set; } = "";
        public int Anio { get; set; }
        public double AutonomiaKm { get; set; }
        public DateTime FechaRegistro { get; set; }
        public Bateria Bateria { get; set; } = new();
    }

    public class AutoHibrido
    {
        public int Id { get; set; }
        public string Marca { get; set; } = "";
        public string Modelo { get; set; } = "";
        public int Anio { get; set; }
        public double AutonomiaKm { get; set; }
        public DateTime FechaRegistro { get; set; }
        public double ConsumoCombustibleL100km { get; set; }
        public double CapacidadBateriaKwh { get; set; }
    }
}
